### ISSUE TITLE
Scope of features var in template

### ISSUE THREAD

**Issue by dfarrell07** [dfarrell07/puppet-opendaylight#40]
A downstream consumer of this Puppet mod reported issues with the scope of the `@features` var in the Karaf config template. Although he didn't have many details, his guess was that because he was consuming the ODL mod along with many others, the tests local to the ODL mod weren't picking up the issue. I added a ton of relevant Beaker tests in 082f0d6e05105ce995c10e5f40e5389aba1c94b9 to try to highlight the issue (because TDD), but was unable to do so. Reading the docs, I did find [this](https://docs.puppetlabs.com/guides/templating.html#out-of-scope-variables) potentially relevant. Using the `scope.lookupvar('opendaylight::features').join(",")` format doesn't break anything for the mod's rspec-puppet or Beaker tests, but I'm waiting on feedback from the other dev to know if it fixed the issue in their more complex environment.

**Comment by dfarrell07** [comment]
Per the downstream dev, 68ddaf8671731a1f32a5775312fff0fb729b34f9 fixed it.
